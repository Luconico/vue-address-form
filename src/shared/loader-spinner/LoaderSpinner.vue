<template>
  <div
    class="loader"
    v-bind:style="{ width: size, height: size, 'border-top-color': color }"
  ></div>
</template>

<script>
export default {
  props: {
    color: {
      type: String,
      default: "gray",
    },
    size: {
      type: String,
      default: "40px",
    },
  },
};
</script>

<style scoped>
.loader {
  display: inline-block;
  border: 3px solid rgba(255, 255, 255, 0.3);
  border-radius: 50%;
  animation: spin 1s ease-in-out infinite;
  -webkit-animation: spin 1s ease-in-out infinite;
}

@keyframes spin {
  to {
    -webkit-transform: rotate(360deg);
  }
}
@-webkit-keyframes spin {
  to {
    -webkit-transform: rotate(360deg);
  }
}
</style>