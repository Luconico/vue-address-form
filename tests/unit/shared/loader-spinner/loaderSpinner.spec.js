import LoaderSpinner from '@/shared/loader-spinner/LoaderSpinner.vue';
import { shallowMount } from '@vue/test-utils';

describe('LoaderSpinner.vue', () => {

    const wrapper = shallowMount(LoaderSpinner, {
        props: {
            size: '20px',
            color: 'red'
        }
    });
  it('Should match whit snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('Should props size and color', () => {
    expect(wrapper.props()).toEqual({
        size: '20px',
        color: 'red'
    });
  });

});