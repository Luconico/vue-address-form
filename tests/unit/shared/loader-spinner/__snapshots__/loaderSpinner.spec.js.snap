// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoaderSpinner.vue Should match whit snapshot 1`] = `<div class="loader" style="width: 20px; height: 20px; border-top-color: red;"></div>`;
